promote/common_utils.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+exit_if_not_on_git_tag() {
+    # Have an override for debugging purposes
+    if [[ -n "${TEST_WITHOUT_GIT_TAG-}" ]] ;then
```

Comment:
So, I don't personally write bash scripts with `set -u`, so this is new to me. However, it feels like slapping on the trailing hyphen on every variable usage is defeating the point of `set -u` (which is to tell you if you've, e.g., mispelled an env var). Wouldn't it be better to maintain the invariant that `TEST_WITHOUT_GIT_TAG` is always set in your scripts (by setting it if it is not set), and then using the non-hyphenated form?